### PR TITLE
introduce lapc and spc debug_ip sub device

### PR DESCRIPTION
**Pull request by chienwei-lan**

### CODE REVIEW

File: src/runtime_src/core/pcie/driver/linux/xocl/subdev/lapc.c [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+
+static int lapc_close(struct inode *inode, struct file *file)
+{
+	struct xocl_lapc *lapc = file->private_data;
+
+	xocl_drvinst_close(lapc);
+	return 0;
+}
+
+long lapc_ioctl(struct file *filp, unsigned int cmd, unsigned long arg)
```
Comment (houlz0507) [inline src/runtime_src/core/pcie/driver/linux/xocl/subdev/lapc.c:116]: Why ioctl been implemented by actually doing nothing?

File: src/runtime_src/core/pcie/driver/linux/xocl/subdev/icap.c [hunk 2]
```diff
@@ -2134,7 +2142,7 @@ static int icap_verify_bitstream_axlf(struct platform_device *pdev,
 
 done:
 	if (err)
-		xocl_dyn_subdevs_destory(xdev);
+		xocl_subdev_destroy_by_level(xdev, XOCL_SUBDEV_LEVEL_URP);
```
Comment (houlz0507) [inline src/runtime_src/core/pcie/driver/linux/xocl/subdev/icap.c:2145]: I think we can remove this call. At this moment, the xclbin is already downloaded to hardware. We can keep the subdevs been created successfully. They will be removed when download next xclbin, switch PLP, etc.

File: src/runtime_src/core/pcie/driver/linux/xocl/subdev/lapc.c (hunk 1 above)
Comment (hackwa) [reply src/runtime_src/core/pcie/driver/linux/xocl/subdev/lapc.c:116]: Hey @houlz0507 the actual ioctl calls will be implemented by profile/debug team.

File: src/runtime_src/core/pcie/driver/linux/xocl/subdev/icap.c (hunk 2 above)
Comment (chienwei-lan) [reply src/runtime_src/core/pcie/driver/linux/xocl/subdev/icap.c]: done

File: src/runtime_src/core/pcie/driver/linux/xocl/subdev/lapc.c (hunk 1 above)
Comment (IshitaGhosh) [reply src/runtime_src/core/pcie/driver/linux/xocl/subdev/lapc.c:116]: We may not even need to implement ioctls for lapc and spc as XDP library does not read/write to LAPC/SPC monitors directly. LAPC/SPC are only read by "xbutil". We can just implement/use sysfs nodes for LAPC, SPC for xbutil.

**Pull request merged**